### PR TITLE
HPCC-23043 CRoxieServerSoapDatasetActionActivity does not check for re-execution

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Compiled. Smoketest should through up any major regressions, but actually testing the issue is tricky.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday

File: roxie/ccd/ccdserver.cpp [hunk 1]
```diff
@@ -26987,9 +26987,50 @@ class CRoxieServerSoapActivityBase : public CRoxieServerActivity, implements ISo
...
+        : CRoxieServerSoapActivityBase(_ctx, _factory, _probeManager)
+    {
+    }
+
+    virtual void execute(unsigned parentExtractSize, const byte * parentExtract)
+    {
+        CriticalBlock b(ecrit);
+        if (exception)
+            throw(exception.getLink());
+        if (!executed)
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp:27003]: executed is never set to true!

File: roxie/ccd/ccdserver.cpp [hunk 2]
```diff
@@ -26987,9 +26987,50 @@ class CRoxieServerSoapActivityBase : public CRoxieServerActivity, implements ISo
...
+        {
+            try
+            {
+                onExecute(parentExtractSize, parentExtract);
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp:27007]: this isn't call start/stop() unlike CRoxieServerInternalSinkActivity.  Is that deliberate.  I think that may mean dependencies will not be executed.

**Review by ghalliday — COMMENTED** [review]
@richardkchapman I think there are a couple of problems.

File: roxie/ccd/ccdserver.cpp (hunk 2 above)
Comment (richardkchapman) [reply roxie/ccd/ccdserver.cpp:27007]: It's all a bit of a mess - the start/stop are in one (but not both) of the onExecutes being called...

I suspect that means that one of them had a bug previously...

**Comment by richardkchapman** [comment]
@ghalliday Please re-review

**Review by ghalliday — COMMENTED** [review]
@richardkchapman looks good.  Please squash.

**Comment by richardkchapman** [comment]
@ghalliday Squashed

**Pull request merged**